Marlin/src/module/planner.cpp

```diff
@@ -774,6 +779,21 @@ block_t* Planner::get_current_block() {
 
   return nullptr;
 }
+block_t* Planner::lookahead(uint8_t offset) {
```

Comment:
I think there's a potential race condition associated with this function. It looks ahead to blocks which still may be subject to change due to recalculation of the trapezoidal profiles. And since it is going to be called from within an ISR context the block's parameters may actually be in an inconsistent state when the ISR queries it.

Reply:
The inconsistent state should be avoided by the `if (block->flag.recalculate) return nullptr;` line.

You are right about one thing tho: since the fn doesn't mark the future block as busy, it is possible that the block ends up changing before it is executed, in which case the pressure target will have been wrong. This is not very problematic because:
* The smoothing filter will gracefully and smoothly start aiming at the corrected target without abrupt rate changes (no skipped steps)
* The lookahead necessary to keep extruder accel < 3k is typically around 10ms for direct drive, and 80ms for bowden, so only the beginning of the next block will be looked at.

Reply:
Oh, yeah, that makes sense. The whole thing is in ISR context including the `block->flag.recalculate` guard.

Reply:
To be honest this was the only thing I was worried about so far. Other comments have all been about code readability.